### PR TITLE
feat: improve HTML documentation

### PR DESCRIPTION
**Pull request by sockmaster27**
Milestone 2

Related to #6327

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Feel free to post a screen shot when there is something ready

**Comment by sockmaster27** [comment]
image

**Comment by sockmaster27** [comment]
Personally, I really like this design:
image
image

**Comment by magnus-madsen** [comment]
I think it looks good. We should probably have these categories:

- Modules
- Classes
- Effects
- Types
- Type Aliases
- Definitions

Did I forget anything? 

I would try to port the colors from: api.flix.dev (i.e. the red, green, etc symbols).

**Comment by magnus-madsen** [comment]
Looks great so far!

A few suggestions for improvements (I understand this is still a WIP):

- [ ] Use small icons/glyps of different colors for different entities (see api.flix.dev).
- [ ] Bold the "Modules", "Definitions" etc. headers.
- [ ] Decrease the font size of the list elements. 
- [ ] Use a code font for the list elements (not headings).
- [ ] Clicking on "Definitions" could jump to the top of where definitions start on the page.
- [ ] Sort everything.

**Comment by sockmaster27** [comment]
We should probably think about whether or not we actually want ligatures in the documentation -- On one hand they make it easier to read, but on the other, they might genuinely hurt communication. 

For example:
image
vs.
image

There are compromises of course, like only disabling them in names.

**Comment by magnus-madsen** [comment]
!!!jar

**Comment by magnus-madsen** [comment]
Looks like I broke something with the `doc` command.

**Comment by sockmaster27** [comment]
Works fine for me?

**Comment by magnus-madsen** [comment]
Its my bad :)

**Comment by magnus-madsen** [comment]
TLDR: If the input program had an error nothing would be printed.

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
Excellent. Upwards and onwards 🚀
